crates/transaction-pool/src/traits.rs

```diff
@@ -753,6 +753,33 @@ impl fmt::Display for CanonicalStateUpdate<'_> {
     }
 }
 
+/// Helper error type representing different reasons why a transaction might fail during
+/// processing in the transaction pool.
+#[derive(Debug, derive_more::Display, derive_more::Error)]
+pub enum InvalidKind {
+    /// The type of gas limit that was exceeded (block or blob)
+    #[display("{kind} exceeds limit: required {required}, limit {limit}")]
+    ExceedsGasLimit {
+        /// Type of gas limit exceeded
+        kind: GasLimitKind,
+        /// The amount of gas required by the transaction
+        required: u64,
+        /// The maximum gas limit allowed
+        limit: u64,
+    },
+    /// Represents consensus-level validation errors that can occur when validating a transaction.
+    InvalidTransaction(InvalidTransactionError),
+}
+
+/// Specifies which type of gas limit was exceeded
+#[derive(Debug, Clone, Copy, derive_more::Display)]
+pub enum GasLimitKind {
+    /// Block gas limit exceeded
+    Block,
+    /// Blob gas limit exceeded
+    Blob,
+}
+
```

Comment:
I created this enum as suggested in the issue's description, yet why not reusing the [PoolErrorKind](https://github.com/paradigmxyz/reth/blob/main/crates/transaction-pool/src/error.rs#L41) enum instead @mattsse and adding some new variants ?

Reply:
I actually think `InvalidPoolTransactionError `https://github.com/paradigmxyz/reth/blob/404f8f877842efa887a1b39e78de2eb6ec97289d/crates/transaction-pool/src/error.rs#L189 might be better?

Reply:
I replaced the created enum by using directly `InvalidPoolTransactionError`